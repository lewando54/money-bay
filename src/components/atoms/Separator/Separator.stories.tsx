/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import Separator from './Separator'

const meta = {
	component: Separator,
	argTypes: {
        // To write
	}
} satisfies Meta<typeof Separator>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
	args: {
        // To write
	}
}
