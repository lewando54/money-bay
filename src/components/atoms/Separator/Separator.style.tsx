import { StyleSheet } from 'react-native'
export default StyleSheet.create({
    mainStyle: {
        backgroundColor: '#CED6E1', 
        width: '100%', 
        height: 1
    }
})
