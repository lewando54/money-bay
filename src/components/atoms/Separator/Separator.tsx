import React from 'react'
import { StyleProp, View, ViewStyle } from 'react-native'
import separatorStyle from './Separator.style'

interface ISeparatorProps {
    style?: StyleProp<ViewStyle>
}

export default function Separator({
    style
}: ISeparatorProps){
    return (
        <View style={[separatorStyle.mainStyle, style]}></View>
    )
}
