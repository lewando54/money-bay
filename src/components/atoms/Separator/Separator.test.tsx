import React from 'react'
import renderer from 'react-test-renderer'
import { render, screen } from '@testing-library/react-native'
import Separator from './Separator'
describe('Separator', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Separator />)
    expect(tree).toMatchSnapshot()
  })
})
