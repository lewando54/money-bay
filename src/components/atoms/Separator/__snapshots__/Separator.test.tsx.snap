// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator should render correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#CED6E1",
        "height": 1,
        "width": "100%",
      },
      undefined,
    ]
  }
/>
`;
